FeedSX/screens/createPost/index.tsx

```diff
@@ -0,0 +1,549 @@
+        </View>
+        {/* text input field */}
+        <LMInputText
+          placeholderText="Write something here..."
+          placeholderTextColor="#0F1E3D66"
+          inputTextStyle={{
+            marginHorizontal: 15,
+            marginVertical: 8,
+            fontSize: 16,
+            elevation: 0,
+          }}
+          multilineField
+          inputText={postContentText}
+          onType={val => {
+            setPostContentText(val);
+            detectLinks(postContentText);
+          }}
+        />
+        {/* selected media section */}
+        <View>
+          {/* multi media selection section */}
+          {formattedMediaAttachments ? (
+            formattedMediaAttachments?.length > 1 ? (
+              <LMCarousel
+                attachments={formattedMediaAttachments}
+                showCancel
+                onCancel={index => removeMediaAttachment(index)}
+              />
+            ) : (
+              <>
+                {/* single image selected section */}
+                {formattedMediaAttachments[0]?.attachmentMeta.format ===
+                  'image/jpg' && (
+                  <LMImage
+                    imageUrl={`${formattedMediaAttachments[0]?.attachmentMeta.url}`}
+                    showCancel
+                    onCancel={() => removeSingleAttachment()}
+                  />
+                )}
+                {/* single video selected section  */}
+                {formattedMediaAttachments[0]?.attachmentMeta.format ===
+                  'video/mp4' && (
+                  <LMVideo
+                    videoUrl={`${formattedMediaAttachments[0]?.attachmentMeta.url}`}
+                    showCancel
+                    onCancel={() => removeSingleAttachment()}
+                  />
+                )}
+              </>
+            )
+          ) : null}
+          {/* selected document view section */}
+          {formattedDocAttachments && formattedDocAttachments.length >= 1 && (
+            <LMDocument
+              attachments={formattedDocAttachments}
+              showCancel
+              onCancel={index => removeDocAttachment(index)}
+            />
+          )}
+          {/* added link preview section */}
+          {formattedMediaAttachments.length <= 0 &&
+            formattedDocAttachments.length <= 0 &&
+            formattedLinkAttachments &&
+            formattedLinkAttachments.length >= 1 && (
+              <LMLinkPreview
+                attachments={formattedLinkAttachments}
+                showCancel
+                onCancel={() => {
+                  setFormattedLinkAttachments([]);
+                }}
+              />
+            )}
+        </View>
+        {/* add more media button section */}
+        {(formattedMediaAttachments.length > 0 ||
+          formattedDocAttachments?.length > 0) && (
+          <LMButton
+            onTap={
+              formattedMediaAttachments.length > 0
+                ? () => handleGallery('mixed')
```

Comment:
create enums for `mixed`, `image` and `video` and then use for passing.

Reply:
resolved